tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,42 @@
  * @{
  *
  * @file
- * @brief       Test application for the LIS2DH12 accelerometer driver
+ * @brief       Test application for LIS2DH12 accelerometer driver
  *
- * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Jan Mohr <jan.mohr@ml-pa.com>
  *
  * @}
  */
 
-#include <assert.h>
-#include <stdio.h>
+#include "shell.h"
 
-#include "fmt.h"
-#include "xtimer.h"
-#include "mutex.h"
-#include "lis2dh12.h"
-#include "lis2dh12_params.h"
+#include "thread.h"
 
- /* delay between sensor data reads */
-#define DELAY       (100UL * US_PER_MS)
+#include "include/lis2dh12_test.h"
 
-/* allocate some memory to hold one formatted string for each sensor output, so
- * one string for the X, Y, and Z reading, respectively */
-static char str_out[3][8];
+kernel_pid_t lis2dh12_process;
 
-/* allocate device descriptor */
-static lis2dh12_t dev;
-
-#ifdef MODULE_LIS2DH12_INT
-/* control interrupt */
-typedef struct {
-    uint8_t line;
-    mutex_t *lock;
-    uint8_t *flags;
-} lis_ctx;
-
-/* timer lock */
-static uint8_t isr_flags;
-static mutex_t isr_mtx = MUTEX_INIT_LOCKED;
-static lis_ctx ctx[2] = {
-    {
-        .line  = 1,
-        .lock  = &isr_mtx,
-        .flags = &isr_flags,
-    }, {
-        .line  = 2,
-        .lock  = &isr_mtx,
-        .flags = &isr_flags,
-    }
+static const shell_command_t shell_commands[] = {
+                { "lis", "Command with multiple subcommands.", shell_lis2dh12_cmd },
+                { NULL, NULL, NULL },
 };
 
-/* interrupt callback function. */
-static void lis2dh12_int_cb(void* _ctx) {
-    lis_ctx *control = _ctx;
-
-    *control->flags |= control->line;
-
-    mutex_unlock(control->lock);
-}
-
-/* print interrupt register */
-static void lis2dh12_int_reg_content(lis2dh12_t *dev, uint8_t pin){
-
-    assert(pin == LIS2DH12_INT1 || pin == LIS2DH12_INT2);
-
-    uint8_t buffer;
-    lis2dh12_read_int_src(dev, &buffer, pin);
-
-    printf("content SRC_Reg_%d: 0x%02x\n", pin, buffer);
-    printf("\t XL %d\n", !!(buffer & LIS2DH12_INT_SRC_XL));
-    printf("\t XH %d\n", !!(buffer & LIS2DH12_INT_SRC_XH));
-    printf("\t YL %d\n", !!(buffer & LIS2DH12_INT_SRC_YL));
-    printf("\t YH %d\n", !!(buffer & LIS2DH12_INT_SRC_YH));
-    printf("\t ZL %d\n", !!(buffer & LIS2DH12_INT_SRC_ZL));
-    printf("\t ZH %d\n", !!(buffer & LIS2DH12_INT_SRC_ZH));
-    printf("\t IA %d\n", !!(buffer & LIS2DH12_INT_SRC_IA));
-}
-#endif
+char lis2dh12_process_stack[THREAD_STACKSIZE_MAIN];
 
 int main(void)
 {
+    /* processing lis2dh12 acceleration data */
+    lis2dh12_process = thread_create(lis2dh12_process_stack, sizeof(lis2dh12_process_stack),
+                  THREAD_PRIORITY_MAIN - 1, THREAD_CREATE_SLEEPING,
+                  lis2dh12_test_process, NULL, "lis2dh12_process");
```

Comment:
If you move the thread creation into `lis2dh12_test_init()` you don't have to share the PID across files.